core/runtime/src/main/java/io/quarkiverse/langchain4j/RegisterAiService.java

```diff
@@ -140,6 +145,11 @@
      */
     Class<? extends Supplier<ModerationModel>> moderationModelSupplier() default BeanIfExistsModerationModelSupplier.class;
 
+    /**
+     * Configures a toolProvider. Either a toolProvider or "normal" tools can be used, but not both
+     */
+    Class<? extends ToolProvider> toolProvider() default BeanIfExistsToolProviderSupplier.class;
```

Comment:
Is there a reason this doesn't follow the supplier model of all the other configuration options (for example for `AuditService`)? 

The reason this model is nice is that it allows advanced use cases to create their own implementations, so my suggestion would be to follow it.